spring-pulsar/src/main/java/org/springframework/pulsar/core/DefaultTopicResolver.java

```diff
@@ -91,6 +94,9 @@ private Optional<String> doResolveTopic(@Nullable String userSpecifiedTopic, @Nu
 		if (StringUtils.hasText(userSpecifiedTopic)) {
 			return Optional.of(userSpecifiedTopic);
 		}
+		if (messageType == null) {
```

Comment:
This change will break [this about to merged change](https://github.com/spring-projects-experimental/spring-pulsar/pull/322/files#diff-60aaf1bc942ff45a5fa56d82ad06cfe23c20b4ffa9d1dd9f2d23a2818b969677R98). 

I switched to use the TR in the sender/producer so that all of the flows go through the resolver. If the user changed the TR impl then it would be nice if all flows go through there. The sender/producer do not have the message on their own at this point so the default impl just does `userTopic OR defaultTopic` resolution. In this proposed changed that would instead do `userTopic or empty`.